pkg/client/core/state/storm/cluster_state.go

```diff
@@ -41,18 +42,38 @@ func (c *Cluster) Convert() *client.Cluster {
 }
 
 func (c *clusterState) SaveCluster(cluster *client.Cluster) error {
-	return c.node.Save(NewCluster(cluster, NewMetadata()))
+	existing, err := c.getCluster(cluster.Name)
+	if err != nil && !errors.Is(err, stormpkg.ErrNotFound) {
+		return err
+	}
+
+	if errors.Is(err, stormpkg.ErrNotFound) {
+		return c.node.Save(NewCluster(cluster, NewMetadata()))
```

Comment:
Is there a reason NewCluster is exported? (And the other New* functions in this package?)

Reply:
They could probably be package scoped, but there isn't any real reason to conceal them, they aren't that internal. Also, I was thinking I might use them for the migration stuff, TBD